api/base/serializers.py

```diff
@@ -267,7 +267,7 @@ def __init__(self, **kwargs):
 
     def to_internal_value(self, data):
         if self.target_type != data:
-            raise Conflict()
+            raise Conflict(detail=('The resource type you specified "' + data + '" does not match the type of the target resource you specified "' + self.target_type + '".'))
```

Comment:
And since this is a relationship field, if you know which relationship it belongs to, we can probably make it even better.